maven-artifact/src/main/java/org/apache/maven/artifact/Artifact.java

```diff
@@ -86,6 +87,16 @@
 
     void setFile( File destination );
 
+    default Path getPath()
+    {
+        return this.getFile().toPath();
```

Comment:
If `getFile()` is `null`, it should not throw an NPE

Reply:
Good catch! Just fixed it using force-push. If you prefer, I can replace by `Optional`.